### PR TITLE
add status code checks in unit tests

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
I missed several status code checks with `IotHubClientException` in our unit tests, so add them back.

### CODE REVIEW

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs [hunk 1]
```diff
@@ -221,14 +221,18 @@ public async Task DeviceNotFoundExceptionReturnsDeviceDisabledStatus()
...
 
             // act
-            await ((Func<Task>)(() => sut
-                .OpenAsync(CancellationToken.None)))
-                .ExpectedAsync<IotHubClientException>()
-                .ConfigureAwait(false);
+            Func<Task> act = async () =>
+            {
+                await sut
+                .OpenAsync(CancellationToken.None).ConfigureAwait(false);
```
Comment (drwill-ms) [inline iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs:227]: Line length not too long to require breaking
```suggestion
                await sut.OpenAsync(CancellationToken.None).ConfigureAwait(false);
```

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs [hunk 2]
```diff
@@ -221,14 +221,18 @@ public async Task DeviceNotFoundExceptionReturnsDeviceDisabledStatus()
...
+            };
 
-            // assert
+            // assert for connection status
             connectionStatusInfo.Status.Should().Be(ConnectionStatus.Disconnected);
             connectionStatusInfo.ChangeReason.Should().Be(ConnectionStatusChangeReason.DeviceDisabled);
+            // assert for exception
+            var error = await act.Should().ThrowAsync<IotHubClientException>();
+            error.And.StatusCode.Should().Be(IotHubStatusCode.DeviceNotFound);
```
Comment (drwill-ms) [inline iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs:235]: Does this work?

```suggestion
            await act.Should().ThrowAsync<IotHubClientException>()
                .And.StatusCode.Should().Be(IotHubStatusCode.DeviceNotFound);
```

**Review by drwill-ms — APPROVED** [review]

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs (hunk 2 above)
Comment (brycewang-microsoft) [reply iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs:235]: No, I tested this previously but somehow it didn't

File: iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs (hunk 1 above)
Comment (brycewang-microsoft) [reply iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs:227]: Yeah, I missed it

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**